src/main/java/by/exadel/internship/service/impl/InterviewServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+import by.exadel.internship.dto.time_for_call.UserTimeSlotDTO;
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserIdDTO;
+import by.exadel.internship.entity.Interview;
+import by.exadel.internship.exception_handing.InappropriateRoleException;
+import by.exadel.internship.mapper.InterviewMapper;
+import by.exadel.internship.repository.InterviewRepository;
+import by.exadel.internship.service.FormService;
+import by.exadel.internship.service.InterviewService;
+import by.exadel.internship.service.UserService;
+import by.exadel.internship.service.UserTimeSlotService;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class InterviewServiceImpl implements InterviewService {
+
+    private static final String SIMPLE_CLASS_NAME = InterviewService.class.getSimpleName();
+
+    private final InterviewRepository interviewRepository;
+    private final InterviewMapper mapper;
+    private final UserTimeSlotService userTimeSlotService;
+    private final FormService formService;
+    private final UserService userService;
+
+
+    @Override
+    public List<InterviewDTO> getAllByUserId(UUID userId, UserRole userRole) {
+        List<Interview> interviews;
+        switch (userRole) {
+            case ADMIN: {
+                interviews = interviewRepository.findAllByAdmin(userId);
+                break;
+            }
+            case TECH_EXPERT: {
+                interviews = interviewRepository.findAllByTechSpecialist(userId);
+                break;
+            }
+            default: {
+                throw new InappropriateRoleException("Inappropriate role", "user.role.invalid");
+            }
+        }
+        return mapper.mapToInterviewDTO(interviews);
+    }
+
+
+    @Override
+    public void saveInterviewForForm(UUID formId, InterviewSaveDTO interviewDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save Form with Interview");
+        FormFullDTO formFullDTO = formService.getById(formId);
+        UserDTO userDTO = userService.getById(interviewDTO.getUserId());
+        if (userDTO.getUserRole() == UserRole.ADMIN && formFullDTO.getFormStatus() == FormStatus.REGISTERED) {
+            InterviewDTO interviewFullDTO = new InterviewDTO();
+            interviewFullDTO.setAdmin(interviewDTO.getUserId());
+            interviewFullDTO.setAdminInterviewDate(interviewDTO.getUserInterviewDate());
+
+            deleteTime(interviewDTO.getUserInterviewDate(), interviewDTO.getUserId());
+
+            formFullDTO.setFormStatus(FormStatus.ADMIN_INTERVIEW_ASSIGNED);
+            formService.updateForm(formFullDTO);
+        }else {
+            if (formFullDTO.getFormStatus() != FormStatus.REGISTERED) {
```

Comment:
Move this checks before saving. It helps you to avoid double checking.

Reply:
I did double check for understand which exception i should throw

Reply:
It's minor change. Let's do it in another PR.